pkgs/development/python-modules/torch-geometric/default.nix

```diff
@@ -0,0 +1,152 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  flit-core,
+
+  # dependencies
+  aiohttp,
+  fsspec,
+  jinja2,
+  numpy,
+  psutil,
+  pyparsing,
+  requests,
+  torch,
+  tqdm,
+
+  # optional-dependencies
+  matplotlib,
+  networkx,
+  pandas,
+  protobuf,
+  wandb,
+  ipython,
+  matplotlib-inline,
+  pre-commit,
+  torch-geometric,
+  ase,
+  # captum,
+  graphviz,
+  h5py,
+  numba,
+  opt-einsum,
+  pgmpy,
+  pynndescent,
+  # pytorch-memlab,
+  rdflib,
+  rdkit,
+  scikit-image,
+  scikit-learn,
+  scipy,
+  statsmodels,
+  sympy,
+  tabulate,
+  torchmetrics,
+  trimesh,
+  pytorch-lightning,
+  yacs,
+  huggingface-hub,
+  onnx,
+  onnxruntime,
+  pytest,
+  pytest-cov,
+}:
+
+buildPythonPackage rec {
+  pname = "torch-geometric";
+  version = "2.6.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "pyg-team";
+    repo = "pytorch_geometric";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Zw9YqPQw2N0ZKn5i5Kl4Cjk9JDTmvZmyO/VvIVr6fTU=";
+  };
+
+  build-system = [
+    flit-core
+  ];
+
+  dependencies = [
+    aiohttp
+    fsspec
+    jinja2
+    numpy
+    psutil
+    pyparsing
+    requests
+    torch
+    tqdm
+  ];
+
+  optional-dependencies = {
+    benchmark = [
+      matplotlib
+      networkx
+      pandas
+      protobuf
+      wandb
+    ];
+    dev = [
+      ipython
+      matplotlib-inline
+      pre-commit
+      torch-geometric
+    ];
+    full = [
+      ase
+      # captum
+      graphviz
+      h5py
+      matplotlib
+      networkx
+      numba
+      opt-einsum
+      pandas
+      pgmpy
+      pynndescent
+      # pytorch-memlab
+      rdflib
+      rdkit
+      scikit-image
+      scikit-learn
+      scipy
+      statsmodels
+      sympy
+      tabulate
+      torch-geometric
+      torchmetrics
+      trimesh
+    ];
+    graphgym = [
+      protobuf
+      pytorch-lightning
+      yacs
+    ];
+    modelhub = [
+      huggingface-hub
+    ];
+    test = [
```

Comment:
Are we running the tests?

Reply:
Now yes.
I didn't add `onnx` and `onnxruntime` to the `nativeCheckInputs` as it made no difference.